### DISCUSSION TITLE
Generated Spec not matching defined resource

### DISCUSSION THREAD

**Discussion by keepevets** [tailflow/laravel-orion#258] [Bugs]
I'm curious about the spec builder, 

I have an api set up that works, for my controllers i'm defining a protected $resource and that resource is set to return specific fields, however whenever i use the spec generation tool 'php artisan orion:specs`, the generated spec shows schema directly from the database columns instead of what i have limited in the resource class for the object, is this the intended behavior? shouldn't the spec generation only include the data that i'm returning in the resource class?

**Comment by keepevets** [comment]
edit: the api does only return the specified data, however the spec shows more columns than are actually returned and i believe this is confusing since the spec doesnt actually match what is returned by the api

**Comment by alberto-bottarini** [comment]
This is a big weakness in my opinion. Resources are a must-have to hide private columns.

**Comment by vintalie** [comment]
I'm having a problem running orion:specs; it gives me an error because it tries to build a path that doesnt actually exist (search).

I think it has something to do with building based on the database and not the controller...

<img width="1266" height="699" alt="imagem" src="https://github.com/user-attachments/assets/c4b20451-80ce-4945-b4cf-2d7a681a9551" />

that is the big villain, i guess

<img width="657" height="370" alt="imagem" src="https://github.com/user-attachments/assets/aa9894c0-04f5-458c-a4ba-ad060850f2bb" />

the cache is going with "search" path, or something else? pls give me a light kakakka
